include/boost/geometry/algorithms/detail/convex_hull/interface.hpp

```diff
@@ -88,17 +86,18 @@ struct hull_to_geometry
     static inline void apply(Geometry const& geometry, OutputGeometry& out,
             Strategy const& strategy)
     {
+        // TODO: Why not handle multi-polygon here?
+        // TODO: detail::as_range() is only used in this place in the whole library
+        //       it should probably be located here.
+        // NOTE: A variable is created here because this can be a proxy range
+        //       and back_insert_iterator<> can store a pointer to it.
+        // Handle linestring, ring and polygon the same:
+        auto&& range = detail::as_range(out);
```

Comment:
I am not sure if `auto` is of good use here since it is difficult to derive the type of `range`

Reply:
What do you mean by "it is difficult to derive the type of `range`"?

Do you mean that you'd prefer the following? Or do you have something else in mind?
```
typename ring_return_type<Geometry>::type range = detail::as_range(out);
```

Btw, I'm working on a different PR in which this will be removed.

Reply:
> What do you mean by "it is difficult to derive the type of `range`"?
> 
> Do you mean that you'd prefer the following? Or do you have something else in mind?
> 
> ```
> typename ring_return_type<Geometry>::type range = detail::as_range(out);
> ```

yes this is what I mean, it is more clear 

> Btw, I'm working on a different PR in which this will be removed.

OK so there should not be any issue then.

Reply:
> OK so there should not be any issue then.

I'm using `auto&&` there too, e.g.:
```
auto&& ring = exterior_ring(polygon);
```
so it's possible that we'll have the same discussion there.

I'll try to explain how I see this and maybe you'll agree. If not please explain why do you think this this is harder to understand because I think otherwise.

In general `typename ring_return_type<Geometry>::type` is generic and can be of arbitrary type. In Boost.Geometry in majority of cases this is a reference but it can be a proxy type since AFAIR this is defined as a trait. However to a person not knowing how Boost.Geometry works on the first sight this looks like a value (not reference). At least I think this is what a random person would think. That this line would be a copy:
```
typename ring_return_type<Geometry>::type range = detail::as_range(out);
```

In the case of universal reference `auto&&` it's clear that range is of generic type and can be a reference or value. So I think that it's more clear than the alternative.